src/shared/configUtils.ts

```diff
@@ -35,27 +48,29 @@ export class ConfigUtils {
     return this.#globalConfig;
   }
 
-  public static async getOrCreateIdentityToken(): Promise<string> {
-    let token = await this.getIdentityToken();
-    if (!token) {
-      token = CryptoUtils.generateIdentityToken();
-      await this.writeIdentityToken(token);
+  public static async getOrCreateIdentityToken(username: string, connection: Connection): Promise<string> {
+    let identityData = await this.getIdentityData();
+    if (!identityData) {
+      const token = CryptoUtils.generateIdentityToken();
+      const entityId = await this.saveIdentityTokenToServer(token, connection);
+      identityData = new LocalWebServerIdentityData(token);
+      identityData.usernameToServerEntityIdMap[username] = entityId;
+      await this.writeIdentityData(identityData);
+      return token;
+    } else {
+      let entityId = identityData.usernameToServerEntityIdMap[username];
+      if (!entityId) {
```

Comment:
There could be a case where in the machine's `sf`'s config the identity data was saved. But, somehow the server doesn't have the identity token saved anymore. In that case maybe I don't check for `entityId` but instead always save back the token to the server. What do you think?

Reply:
Makes sense to me, but I'd defer to Kevin on this.

Reply:
In a world where we don't have the Entity ID for an org stored locally, by definition we no longer have any definitive way to connect back to what may or may not have already existed for this project on the app server. In this case we _have_ to write back to the server, and get a new Entity ID, even if the server has a previous entry for this project. In that case, the previous data becomes orphaned, but otherwise the identity is definitively established with the new entity.

That was an intentional design decision: app server entries needed to potentially be throwaway, because keeping them in sync in all cases would otherwise be too onerous. With this design, you just generate a new association and move on.